libnavigation-core/src/main/java/com/mapbox/navigation/core/MapboxNavigation.kt

```diff
@@ -430,7 +430,7 @@ class MapboxNavigation @VisibleForTesting internal constructor(
             ),
             historyRecorderHandles.composite,
         )
-        val result = MapboxModuleProvider.createModule<Router>(MapboxModuleType.NavigationRouter) {
+        val result = navigationOptions.router ?: MapboxModuleProvider.createModule<Router>(MapboxModuleType.NavigationRouter) {
```

Comment:
What do you mean it's runtime? This code is executed  in `MapboNavigation` constructor and you only use the `result` once to create a `val moduleRouter`. It's not runtime.

Reply:
Thank you. How do suggest me to call this type of configuration then?

Reply:
I'm not sure that's possible right now. I see that `moduleRouter` is used when creating native navigator. And it's immutable there. So we would need NN to make it mutable first. But first let's decide whether we want that at all.